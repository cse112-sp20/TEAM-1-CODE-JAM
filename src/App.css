#side {
  padding-left: 0;
  padding-right: 0;
  background-color: rgb(36, 36, 36);
  /* padding-bottom: %; */
  min-height: 544px;
  margin: 0;
}
