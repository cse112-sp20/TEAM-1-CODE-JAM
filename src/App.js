import React, { Component } from "react";
import { Route, Switch, BrowserRouter as Router } from "react-router-dom";
import CreateJoinTeam from "./components/CreateJoinTeam";
import NavBar from "./components/NavBar";
import "./components/materialize.min.css";
import SideNav from "./components/SideNav";
import "./App.css";
import Teams from "./components/Teams";
import Home from "./components/Home";
import Timeline from "./components/Timeline";

class App extends Component {
  render() {
    return (
      <div className="app">
        <Router>
          {/* NavBar should always be shown for all pages */}
          <NavBar></NavBar>
          <div className="row">
            {/* save spaces for the side navbar */}
            <div className="col s2" id="side">
              <SideNav></SideNav>
            </div>
            {/* This is where the real component is at */}
            <div className="col s10" id="component">
              <Switch>
                {/* follow the following examples for adding new component */}
                <Route
                  exact
                  path="/createjoin"
                  component={CreateJoinTeam}
                ></Route>
                <Route exact path="/teams" component={Teams}></Route>
                <Route exact path="/timeline" component={Timeline}></Route>
                <Route path="/" component={Home}></Route>
              </Switch>
            </div>
          </div>
        </Router>
      </div>
    );
  }
}

export default App;
