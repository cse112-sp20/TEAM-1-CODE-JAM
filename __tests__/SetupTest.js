import '@testing-library/jest-dom'
import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react';
import App from '../src/App';

test('Check that app is set up correctly.', () => {
  render(<App />)

  // query* functions will return the element or null if it cannot be found
  // get* functions will return the element or throw an error if it cannot be found
  // expect(screen.queryByText("Timeline")).toBeNull()

  // the queries can accept a regex to make your selectors more resilient to content tweaks and changes.
  // fireEvent.click(screen.getByLabelText(/show/i))

  // .toBeInTheDocument() is an assertion that comes from jest-dom
  // otherwise you could use .toBeDefined()
  // expect(screen.getByText(testMessage)).toBeInTheDocument()
})
