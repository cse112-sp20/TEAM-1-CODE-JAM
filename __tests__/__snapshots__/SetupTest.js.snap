// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is set up correctly 1`] = `
<div
  className="app"
>
  <div>
    <nav
      id="nav-background"
    >
      <div
        className="nav-wrapper"
      >
        <a
          className="left"
          href="#"
          id="team-name"
        >
          Team Activity Tracker
        </a>
        <ul
          className="right"
        >
          <li>
            <a
              href="#"
              id="adjust-height"
              onClick={[Function]}
            >
              <i
                className="material-icons"
                id="adjust-height"
              >
                settings
              </i>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div
    className="row"
  >
    <div
      className="col s1"
      id="side"
    >
      <div
        className="SideNav"
      >
        <a
          aria-current="page"
          className="btn tooltipped waves-effect waves-light card-panel center-align active"
          data-position="right"
          data-tooltip="Home"
          href="/"
          id="sidenav-background"
          onClick={[Function]}
          style={Object {}}
        >
          <i
            className="large white-text material-icons"
          >
            home
          </i>
        </a>
        <a
          aria-current={null}
          className="btn tooltipped waves-effect waves-light card-panel"
          data-position="right"
          data-tooltip="Timeline"
          href="/timeline"
          id="sidenav-background"
          onClick={[Function]}
        >
          <i
            className="white-text material-icons"
          >
            access_time
          </i>
        </a>
        <a
          aria-current={null}
          className="btn tooltipped waves-effect waves-light card-panel center-align"
          data-position="right"
          data-tooltip="Charts"
          href="/charts"
          id="sidenav-background"
          onClick={[Function]}
        >
          <i
            className="white-text material-icons"
          >
            insert_chart
          </i>
        </a>
        <a
          aria-current={null}
          className="btn tooltipped waves-effect waves-light card-panel center-align"
          data-position="right"
          data-testid="SideNav-teams"
          data-tooltip="Teams"
          href="/teams"
          id="sidenav-background"
          onClick={[Function]}
        >
          <i
            className="white-text material-icons"
          >
            group_add
          </i>
        </a>
      </div>
    </div>
    <div
      className="col s11"
      id="component"
    >
      <div>
        <div
          className="row"
          id="row"
        >
          <div
            className="card blue-grey lighten-5"
          >
            <div
              className="card-content black-text"
              id="card-content"
            >
              <div
                className="row"
                id="top-row"
              >
                <div
                  className="col s4"
                  id="col"
                >
                  <img
                    className="circle"
                    id="profile"
                    src={
                      Object {
                        "chrome": Object {
                          "identity": Object {
                            "getProfileUserInfo": [MockFunction],
                            "getRedirectURL": [MockFunction],
                          },
                        },
                      }
                    }
                  />
                </div>
                <div
                  className="col s8"
                >
                  <span
                    className="truncate card-title black-text center-align"
                    id="title"
                  >
                    
                  </span>
                  <div
                    className="divider"
                  />
                  <div
                    className="row"
                  >
                    <div
                      className="col s4"
                      id="col"
                    >
                      <p
                        className="center-align"
                        id="top"
                      >
                        
                      </p>
                      <p
                        className="center-align blue-grey-text text-darken-1"
                        id="bottom"
                      >
                        Code
                      </p>
                    </div>
                    <div
                      className="col s4"
                      id="col"
                    >
                      <p
                        className="center-align"
                        id="top"
                      >
                        0
                      </p>
                      <p
                        className="center-align dropdown-trigger blue-grey-text text-darken-1"
                        data-target="people"
                        id="bottom"
                      >
                        Members
                      </p>
                      <ul
                        className="dropdown-content collection"
                        id="people"
                      />
                    </div>
                    <div
                      className="col s4"
                      id="col"
                    >
                      <p
                        className="center-align"
                        id="top"
                      >
                        100
                      </p>
                      <p
                        className="center-align blue-grey-text text-darken-1"
                        id="bottom"
                      >
                        Points
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="card-action"
              >
                <div
                  className="row"
                >
                  <div
                    className="col s6"
                    id="col"
                  >
                    <button
                      className="rs-btn rs-btn-ghost tooltipped truncate rs-btn-sm rs-btn-block"
                      data-html="true"
                      data-position="bottom"
                      data-tooltip="Add current website to black list"
                      id="blacklist-button"
                      onClick={[Function]}
                      style={
                        Object {
                          "backgroundColor": "transparent",
                        }
                      }
                      type="button"
                    >
                      unknown website
                    </button>
                  </div>
                  <div
                    className="col s6"
                    id="col"
                  >
                    <div
                      className="switch right"
                      id="checkin"
                    >
                      <label>
                        Check off
                        <input
                          checked={false}
                          onChange={[Function]}
                          type="checkbox"
                        />
                        <span
                          className="lever"
                        />
                        Check in
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="row"
          id="row"
        >
          <div
            className="valign-wrapper card blue-grey lighten-5"
            id="mini-timeline-wrapper"
          >
            <div
              id="mini-timeline"
            >
              <ul
                className="rs-timeline rs-timeline-align-left"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
